// svix/src/openapi/models/CountOut.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { HttpFile } from '../http/http';

export class CountOut {
    /**
    * There\'s a ceiling to how many attempts we count. When the limit is reached, this will be `true` to indicate the actual count is higher than given.
    */
    'approximated': boolean;
    /**
    * The count of attempts matching the query.
    */
    'count': number;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "approximated",
            "baseName": "approximated",
            "type": "boolean",
            "format": ""
        },
        {
            "name": "count",
            "baseName": "count",
            "type": "number",
            "format": "uint64"
        }    ];

    static getAttributeTypeMap() {
        return CountOut.attributeTypeMap;
    }

    public constructor() {
    }
}
